kiosk/src/share.ts

```diff
@@ -0,0 +1,93 @@
+import { GameData } from "./Models/GameData";
+
+const apiRoot = "https://www.makecode.com";
+const description = "A kiosk for MakeCode Arcade";
+
+export interface SharedKioskData {
+    games: GameData[];
+}
+
+export async function createKioskShareLink(kioskData: SharedKioskData) {
+    const payload = createShareRequest(
+        "kiosk",
+        createProjectFiles("kiosk", kioskData)
+    );
+    const url = apiRoot + "/api/scripts";
```

Comment:
So is this how we create all share links across targets? We send a POST request to this endpoint and the payload and header of the request determine what target is used and more?

Reply:
Yeah, here's the entry point for where those get spun up in webapp https://github.com/microsoft/pxt/blob/master/webapp/src/app.tsx#L4106 & the actual requests are made in these two functions depending on if it's anonymous or persistent: https://github.com/microsoft/pxt/blob/master/webapp/src/app.tsx#L4106